medusa/server/api/v2/base.py

```diff
@@ -47,6 +47,9 @@ def api_finish(self, status=None, error=None, data=None, headers=None, **kwargs)
             if data is not None:
                 self.set_header('Content-Type', 'application/json; charset=UTF-8')
                 self.finish(json.JSONEncoder(default=json_string_encoder).encode(data))
+            elif stream:
```

Comment:
@p0psicles this finally fixes the redirect from showPoster issue so it should serve the image instantly. This will help with page loads for larger libraries.